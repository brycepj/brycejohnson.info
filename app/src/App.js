import React, { Component } from "react";

import styled from "styled-components";

import ImageList from "./ImageList";
import ObfuscatedEmail from "./ObfuscatedEmail";

const PageWrap = styled.div``;

const InvitationToContact = styled.p`
  line-height: 25px;
`;

const SeeMoreWrap = styled.div`
  text-align: left;
`;
const SeeLessWrap = styled.div`
  text-align: left;
`;

class App extends Component {
  constructor() {
    super();
    this.state = {
      showMorePictures: false
    };
  }
  showMorePictures = e => {
    e.preventDefault();
    this.setState({ showMorePictures: true });
  };
  showFewerPictures = e => {
    e.preventDefault();
    this.setState({ showMorePictures: false });
  };
  render() {
    return (
      <PageWrap>
        <h2>
          Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.
        </h2>

        <ImageList showMorePictures={this.state.showMorePictures} />
        {!this.state.showMorePictures ? (
          <SeeMoreWrap onClick={this.showMorePictures}>
            <a href="">See a few more...</a>
          </SeeMoreWrap>
        ) : (
          <SeeLessWrap>
            <a href="" onClick={this.showFewerPictures}>
              See a few less...
            </a>
          </SeeLessWrap>
        )}

        <h3>We live on a farm in Virginia with our cats, Ruby and Steve.</h3>

        <h4>
          I work remotely as a senior frontend engineer at{" "}
          <a href="https://www.hirevue.com/">HireVue</a>, and worked previously
          at <a href="https://about.gitlab.com/">GitLab</a> and at two other
          startups.
        </h4>

        <h4>
          In my spare time, I like to make things with code, help with open
          source projects and write about programming. I also enjoy gardening,
          vermiculture (google it!), working on my car, cooking, and listening
          to music.
        </h4>

        <InvitationToContact>
          Feel free to reach out to me through email, at <ObfuscatedEmail />
        </InvitationToContact>
      </PageWrap>
    );
  }
}

export default App;
