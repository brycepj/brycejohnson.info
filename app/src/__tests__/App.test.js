import React from "react";
import { shallow } from "enzyme";
import App from "../App";

it("renders without crashing", () => {
  const result = shallow(<App />);
  expect(result).toMatchSnapshot();
});
