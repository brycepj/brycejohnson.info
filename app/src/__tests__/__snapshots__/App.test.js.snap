// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h2>
          Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.
</h2>,
        <ImageList
          showMorePictures={false}
/>,
        <styled.div
          onClick={[Function]}
>
          <a
                    href=""
          >
                    See a few more...
          </a>
</styled.div>,
        <h3>
          We live on a farm in Virginia with our cats, Ruby and Steve.
</h3>,
        <h4>
          I work remotely as a senior frontend engineer at
           
          <a
                    href="https://www.hirevue.com/"
          >
                    HireVue
          </a>
          , and worked previously at 
          <a
                    href="https://about.gitlab.com/"
          >
                    GitLab
          </a>
           and at two other startups.
</h4>,
        <h4>
          In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.
</h4>,
        <styled.p>
          Feel free to reach out to me through email, at 
          <ObfuscatedEmail />
</styled.p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.",
        },
        "ref": null,
        "rendered": "Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "showMorePictures": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <a
            href=""
>
            See a few more...
</a>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "See a few more...",
            "href": "",
          },
          "ref": null,
          "rendered": "See a few more...",
          "type": "a",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We live on a farm in Virginia with our cats, Ruby and Steve.",
        },
        "ref": null,
        "rendered": "We live on a farm in Virginia with our cats, Ruby and Steve.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "I work remotely as a senior frontend engineer at",
            " ",
            <a
              href="https://www.hirevue.com/"
>
              HireVue
</a>,
            ", and worked previously at ",
            <a
              href="https://about.gitlab.com/"
>
              GitLab
</a>,
            " and at two other startups.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "I work remotely as a senior frontend engineer at",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HireVue",
              "href": "https://www.hirevue.com/",
            },
            "ref": null,
            "rendered": "HireVue",
            "type": "a",
          },
          ", and worked previously at ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "GitLab",
              "href": "https://about.gitlab.com/",
            },
            "ref": null,
            "rendered": "GitLab",
            "type": "a",
          },
          " and at two other startups.",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.",
        },
        "ref": null,
        "rendered": "In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Feel free to reach out to me through email, at ",
            <ObfuscatedEmail />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Feel free to reach out to me through email, at ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h2>
            Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.
</h2>,
          <ImageList
            showMorePictures={false}
/>,
          <styled.div
            onClick={[Function]}
>
            <a
                        href=""
            >
                        See a few more...
            </a>
</styled.div>,
          <h3>
            We live on a farm in Virginia with our cats, Ruby and Steve.
</h3>,
          <h4>
            I work remotely as a senior frontend engineer at
             
            <a
                        href="https://www.hirevue.com/"
            >
                        HireVue
            </a>
            , and worked previously at 
            <a
                        href="https://about.gitlab.com/"
            >
                        GitLab
            </a>
             and at two other startups.
</h4>,
          <h4>
            In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.
</h4>,
          <styled.p>
            Feel free to reach out to me through email, at 
            <ObfuscatedEmail />
</styled.p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.",
          },
          "ref": null,
          "rendered": "Hello, my name is Bryce. I'm the husband of Rachael and dad of Emma.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "showMorePictures": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a
              href=""
>
              See a few more...
</a>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "See a few more...",
              "href": "",
            },
            "ref": null,
            "rendered": "See a few more...",
            "type": "a",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We live on a farm in Virginia with our cats, Ruby and Steve.",
          },
          "ref": null,
          "rendered": "We live on a farm in Virginia with our cats, Ruby and Steve.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "I work remotely as a senior frontend engineer at",
              " ",
              <a
                href="https://www.hirevue.com/"
>
                HireVue
</a>,
              ", and worked previously at ",
              <a
                href="https://about.gitlab.com/"
>
                GitLab
</a>,
              " and at two other startups.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "I work remotely as a senior frontend engineer at",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HireVue",
                "href": "https://www.hirevue.com/",
              },
              "ref": null,
              "rendered": "HireVue",
              "type": "a",
            },
            ", and worked previously at ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "GitLab",
                "href": "https://about.gitlab.com/",
              },
              "ref": null,
              "rendered": "GitLab",
              "type": "a",
            },
            " and at two other startups.",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.",
          },
          "ref": null,
          "rendered": "In my spare time, I like to make things with code, help with open source projects and write about programming. I also enjoy gardening, vermiculture (google it!), working on my car, cooking, and listening to music.",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Feel free to reach out to me through email, at ",
              <ObfuscatedEmail />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Feel free to reach out to me through email, at ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
